clients/include/testing_bsrxmv.hpp

```diff
@@ -0,0 +1,540 @@
+                                                          safe_size,
+                                                          &alpha,
+                                                          descr,
+                                                          dval,
+                                                          dmask_ptr,
+                                                          dptr,
+                                                          dend_ptr,
+                                                          dcol,
+                                                          -1,
+                                                          dx,
+                                                          &beta,
+                                                          dy),
+                                         "Error: block_dim is invalid");
+}
+
+template <typename T>
+hipsparseStatus_t testing_bsrxmv()
+{
+    hipsparseDirection_t dir          = HIPSPARSE_DIRECTION_COLUMN;
+    hipsparseOperation_t trans        = HIPSPARSE_OPERATION_NON_TRANSPOSE;
+    static constexpr int size_of_mask = 1;
+    static constexpr int mb           = 2;
+    static constexpr int nb           = 3;
+    static constexpr int nnzb         = 5;
+    static constexpr int block_dim    = 2;
+
+    T h_alpha = make_DataType<T>(2.0);
+    T h_beta  = make_DataType<T>(1.0);
+
+    std::vector<T>   hbsr_val(block_dim * block_dim * nnzb);
+    std::vector<int> hbsr_mask_ptr(size_of_mask);
+    std::vector<int> hbsr_row_ptr(mb);
+    std::vector<int> hbsr_end_ptr(mb);
+    std::vector<int> hbsr_col_ind(nnzb);
+    std::vector<T>   hx(nb * block_dim);
+    std::vector<T>   hy(mb * block_dim);
+    std::vector<T>   hyref(mb * block_dim);
+
+    auto dbsr_val_managed = hipsparse_unique_ptr{
+        device_malloc(sizeof(T) * block_dim * block_dim * nnzb), device_free};
+    auto dbsr_mask_ptr_managed
+        = hipsparse_unique_ptr{device_malloc(sizeof(int) * size_of_mask), device_free};
+    auto dbsr_row_ptr_managed = hipsparse_unique_ptr{device_malloc(sizeof(int) * mb), device_free};
+    auto dbsr_end_ptr_managed = hipsparse_unique_ptr{device_malloc(sizeof(int) * mb), device_free};
+    auto dbsr_col_ind_managed
+        = hipsparse_unique_ptr{device_malloc(sizeof(int) * nnzb), device_free};
+    auto dx_managed = hipsparse_unique_ptr{device_malloc(sizeof(T) * block_dim * nb), device_free};
+    auto dy_managed = hipsparse_unique_ptr{device_malloc(sizeof(T) * block_dim * mb), device_free};
+
+    T*   dbsr_val      = (T*)dbsr_val_managed.get();
+    int* dbsr_mask_ptr = (int*)dbsr_mask_ptr_managed.get();
+    int* dbsr_row_ptr  = (int*)dbsr_row_ptr_managed.get();
+    int* dbsr_end_ptr  = (int*)dbsr_end_ptr_managed.get();
+    int* dbsr_col_ind  = (int*)dbsr_col_ind_managed.get();
+    T*   dx            = (T*)dx_managed.get();
+    T*   dy            = (T*)dy_managed.get();
+
+    hbsr_mask_ptr[0] = 2;
+
+    hbsr_row_ptr[0] = 1;
```

Comment:
I think you can just do:
std::vector<int> hbsr_row_ptr = {1, 4};

Reply:
I can do that

Reply:
fixed.